service/controller/resource/namespaceauth/resource.go

```diff
@@ -0,0 +1,137 @@
+package namespaceauth
+
+import (
+	"github.com/giantswarm/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	Name = "namespaceauth"
+
+	viewAllRole = "view-all"
+)
+
+type NamespaceAuth struct {
+	ViewAllTargetGroup string
+}
+
+type Config struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+
+	NamespaceAuth NamespaceAuth
+}
+
+type Resource struct {
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+
+	namespaceAuth NamespaceAuth
+}
+
+func New(config Config) (*Resource, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.NamespaceAuth.ViewAllTargetGroup == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.NamespaceAuth.ViewAllTargetGroup must not be empty", config)
+	}
+
+	r := &Resource{
+		k8sClient: config.K8sClient.K8sClient(),
+		logger:    config.Logger,
+
+		namespaceAuth: config.NamespaceAuth,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func appendUnique(slice, exceptions []string, newElement string) []string {
+	for _, e := range slice {
+		for _, exception := range exceptions {
+			if e == newElement || newElement == exception {
+				return slice
+			}
+		}
+	}
+	return append(slice, newElement)
+}
```

Comment:
I think this implementation has a flaw. If `exceptions` is empty then `e == newElement` is never checked so non-unique element can be added. I'd rather turn everything into a `map[string]struct{}` and back to the `[]string` plus iteration over exceptions. Maybe not most efficient but readable and correct.

Reply:
I've dropped exceptions for now as it is not used yet